src/emit/solana/mod.rs

```diff
@@ -1112,12 +1058,33 @@ impl SolanaTarget {
         payload: PointerValue<'b>,
         payload_len: IntValue<'b>,
         contract_args: ContractArgs<'b>,
+        ns: &Namespace,
     ) {
         let instruction_ty: BasicTypeEnum = binary
-            .module
-            .get_struct_type("struct.SolInstruction")
-            .unwrap()
-            .into();
+            .context
+            .struct_type(
+                &[
+                    binary
+                        .module
+                        .get_struct_type("struct.SolPubkey")
+                        .unwrap()
+                        .ptr_type(AddressSpace::default())
+                        .as_basic_type_enum(),
+                    binary
+                        .llvm_type(&Type::Struct(StructType::AccountMeta), ns)
+                        .ptr_type(AddressSpace::default())
+                        .as_basic_type_enum(),
+                    binary.context.i64_type().as_basic_type_enum(),
+                    binary
+                        .context
+                        .i8_type()
+                        .ptr_type(AddressSpace::default())
+                        .as_basic_type_enum(),
+                    binary.context.i64_type().as_basic_type_enum(),
+                ],
+                false,
+            )
+            .as_basic_type_enum();
```

Comment:
declare_externals?

Reply:
I don't think there is a way to add struct definitions to the module.

Reply:
There must be a way to types to a module, how else would you build a complete IR for a file. 

Having said that, I don't know how it is done. I can't find it either